sql/core/src/test/scala/org/apache/spark/sql/connector/MetadataColumnSuite.scala

```diff
@@ -202,12 +203,18 @@ class MetadataColumnSuite extends DatasourceV2SQLBase {
       Seq(sqlQuery, dfQuery).foreach { query =>
         checkAnswer(query, Seq(Row(1, "a", 0, "3/1"), Row(2, "b", 0, "0/2"), Row(3, "c", 0, "1/3")))
       }
+    }
+  }
 
-      assertThrows[AnalysisException] {
-        sql(s"SELECT $sbq.index FROM (SELECT id FROM $tbl) $sbq")
```

Comment:
We disallow it by purpose and I'm not convinced that we should change it.

We can special-case the char padding Project (that only contains Attribute or Alias) in `SubqueryAlias#metadataOutput`